### PR TITLE
add processor boilerplate generator helper

### PR DESCRIPTION
**Pull request by ekneg54**
This adds a little helper for generating code to implement new processors.

### CODE REVIEW

File: logprep/util/template_processor_test.py.j2 [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+        self._load_specific_rule(rule)
+        self.object.process(event)
+        assert event == expected
+
+    @pytest.mark.parametrize("testcase, rule, event, expected", failure_test_cases)
+    def test_testcases_failure_handling(self, testcase, rule, event, expected):
+        self._load_specific_rule(rule)
+        with pytest.raises(ProcessingWarning):
+            self.object.process(event)
+        assert event == expected, testcase
```
Comment (dtrai2) [inline logprep/util/template_processor_test.py.j2:35]:
```suggestion
        assert event == expected, testcase

```

File: logprep/util/processor_generator.py [hunk 2]
```diff
@@ -0,0 +1,114 @@
+"""
+===================
+Processor Generator
+===================
+
+generates boilerplate code to implement a new processor for logprep
+"""
+
+import sys
```
Comment (dtrai2) [inline logprep/util/processor_generator.py:9]:
```suggestion
```

**Review by dtrai2 — CHANGES_REQUESTED** [review]
neat little generator :+1: Could you add it also to the changelog, please.

File: logprep/util/template_processor.py.j2 [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+
+
+class {{ processor.class_name }}({{ processor.base_class.__name__ }}):
+    """A processor that ..."""
+
+    rule_class = {{ processor.class_name }}Rule
+
+    @define(kw_only=True)
+    class Config({{ processor.base_class.__name__}}.Config):
+        ...
```
[the remark this answers is not in the document]
Comment (ekneg54) [reply logprep/util/template_processor.py.j2:33]: done

File: logprep/util/template_rule.py.j2 [hunk 4]
```diff
@@ -0,0 +1,54 @@
+"""
+{{ processor.class_name}}
```
[the remark this answers is not in the document]
Comment (ekneg54) [reply logprep/util/template_rule.py.j2:2]: no... the other processorrules do not do this

File: doc/source/development/processor_how_to.rst [hunk 5]
```diff
@@ -11,12 +11,35 @@ Concurrency/IPC
...
 Processors can run in parallel on multiple different system processes.
 Therefore, it is not guaranteed that a specific processor will see all incoming log messages.
-Inter-process-communication (IPC) must be used if information of log messages has the be shared between multiple processor instances.
+Inter-process-communication (IPC) must be used if information of log messages has to be shared between multiple processor instances.
 IPC is relatively slow and can not be used if the processor instances are located on different machines.
 In those cases it should be reconsidered if it is necessary that information is being shared or if an implementation of the functionality is generally sensible in the context of ths framework.
 
 .. _implementing_a_new_processor:
 
+Getting started
```
[the remark this answers is not in the document]
Comment (ekneg54) [reply doc/source/development/processor_how_to.rst:20]: done

File: logprep/util/template_processor_test.py.j2 [hunk 6]
```diff
@@ -0,0 +1,35 @@
...
+        "type": "{{ processor.name }}",
+        "specific_rules": [],
+        "generic_rules": [],
+    }
+
+    @pytest.mark.parametrize("testcase, rule, event, expected", test_cases)
+    def test_testcases(self, testcase, rule, event, expected):  # pylint: disable=unused-argument
+        self._load_specific_rule(rule)
+        self.object.process(event)
+        assert event == expected
```
[the remark this answers is not in the document]
Comment (ekneg54) [reply logprep/util/template_processor_test.py.j2]: done

File: logprep/util/processor_generator.py [hunk 7]
```diff
@@ -0,0 +1,114 @@
...
+    if isinstance(processor_name, type):
+        return processor_name
+    return Registry.get_class(camel_to_snake(processor_name))
+
+
+@define(kw_only=True)
+class ProcessorCodeGenerator:
+    """Processor generator"""
+
+    name: str = field(validator=validators.instance_of(str), converter=camel_to_snake)
```
[the remark this answers is not in the document]
Comment (ekneg54) [reply logprep/util/processor_generator.py:62]: later ? ;)

[the remark this answers is not in the document]
Comment (ekneg54) [reply logprep/util/processor_generator.py]: done

**Review by dtrai2 — APPROVED** [review]

**Pull request merged**